### ISSUE TITLE
CVEs on current `pedestal/jetty` version

### ISSUE THREAD

**Issue by kelvinqian00** [pedestal/pedestal#744]
# Description

A number of CVEs are present on the most recent `pedestal/jetty` version:
- CVE-2023-26048 and CVE-2023-26049 were detected during NVD scanning about a week or so ago. These CVEs have to do with vulnerabilities in multipart request parsing and cookie parsing, respectively.
- CVE-2023-24998 is a vulnerability in Apache Commons FileUpload that was brought in via Ring.

## Expected Behavior

No CVEs should be present in the project.

## Actual Behavior

NVD scanning using [nvd-clojure](https://github.com/rm-hull/nvd-clojure) following the "Steps to reproduce" reports the following:
```
+---------------------------------------+--------------------------------+
| dependency                            | status                         |
+---------------------------------------+--------------------------------+
| commons-fileupload-1.4.jar            | CVE-2023-24998                 |
| http2-common-9.4.48.v20220622.jar     | CVE-2023-26049, CVE-2023-26048 |
| http2-server-9.4.48.v20220622.jar     | CVE-2023-26049, CVE-2023-26048 |
| jackson-dataformat-cbor-2.9.9.jar     | CVE-2020-28491                 |
| jetty-io-9.4.48.v20220622.jar         | CVE-2023-26049, CVE-2023-26048 |
| jetty-server-9.4.48.v20220622.jar     | CVE-2023-26049, CVE-2023-26048 |
| websocket-client-9.4.48.v20220622.jar | CVE-2023-26049, CVE-2023-26048 |
| websocket-server-9.4.48.v20220622.jar | CVE-2023-26049, CVE-2023-26048 |
+---------------------------------------+--------------------------------+
```
Note that in addition to the above two CVEs, there is also a CVE on Jackson brought in via Cheshire. However, this is essentially an artifact of the pedestal-service version used being `0.5.11-beta-1` instead of `0.6.0` since Cheshire was already updated to the latest version in the latter, addressing that CVE.

## Solution

- Update Jetty dependencies to the latest Jetty 9 version, `9.4.51.v20230217`, similar to what is being done in one of our downstream projects: https://github.com/yetanalytics/lrsql/pull/296.
- Update Ring in `pedestal/service/deps.edn` to `1.10.0`, or apply an `:exclusions` to `commons-fileupload` and use the latest version of _it_ (which is `1.5`).

In addition, it would be helpful going forwards if vulnerability scanning is integrated into Pedestal in order to automatically detect CVEs, rather than relying on contributors detecting them in their downstream applications. One possible method to integrate nvd-clojure in the GitHub Workflow for CI, using a reusable workflow such as [this one](https://github.com/yetanalytics/workflow-nvd). Another option is to create POM file and submit it to [advanced-security/maven-dependency-submission-action](https://github.com/marketplace/actions/maven-dependency-tree-dependency-submission), as we have done in our downstream application [as a GitHub Workflow](https://github.com/yetanalytics/lrsql/blob/main/.github/workflows/deps.yml) that calls a POM-generating [Make command](https://github.com/yetanalytics/lrsql/blob/main/Makefile#L292-L298).

## Steps to reproduce

Install the current version of nvd-clojure as so:
```
clojure -Ttools install nvd-clojure/nvd-clojure '{:mvn/version "RELEASE"}' :as nvd
```

Then in the `jetty` directory of the Pedestal repo, run the following:
```
clojure -Tnvd nvd.task/check :classpath \""$(clojure -Spath)\""
```

# Environment

## Clojure CLI version

1.10.3.1058

## Pedestal version

The reproduction was done on the current commit on the `master` branch, i.e. `0.6.0-beta-1`, but the CVEs were originally detected on our downstream apps when using Pedestal version `0.5.10`.

## nvd-clojure version

3.2.0

**Comment by hlship** [comment]
FYI, to check correctly in the `jetty` folder, the command is: `clojure -Tnvd nvd.task/check :classpath \""$(clojure -Alocal -Spath)\""` (get intra-project dependencies from local, not from Maven).

Made the updates, ran the checks, all clear.

**Comment by hlship** [comment]
We should add to our PR review action, to run these checks.

**Comment by hlship** [comment]
Fixed by #748
